src/script/lua_api/l_util.cpp

```diff
@@ -398,7 +398,8 @@ int ModApiUtil::l_get_dir_list(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	const char *path = luaL_checkstring(L, 1);
-	short is_dir = lua_isboolean(L, 2) ? lua_toboolean(L, 2) : -1;
+	bool list_all = lua_type(L, 2) != LUA_TBOOLEAN; // if its not a boolean list all
```

Comment:
> which would be a bit stupid imo

The macro is there as it is more compact and easier to read.

> If you think !lua_isboolean(L, 2)is better this could be changed while merging.

I could do it for this change, but when you come back and do more changes, it will create more work for me and I don't want that. Can you change it yourself please?